core/shell.py

```diff
@@ -668,6 +661,12 @@ def Main(lang, arg_r, environ, login_shell, loader, readline):
     cmd_ev.MaybeRunExitTrap(box)
     status = box[0]
 
+    if readline:
```

Comment:
One thing to think about is if IOError and OSError that bubble to the top prevent this from being reached

I believe the Interactive() and MaybeRunExitTrap() functions catch MOST exceptions, but not all

But I think we can see if it comes later